server/auto_merge.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v28/github"
+	"github.com/mattermost/mattermost-mattermod/model"
+	"github.com/mattermost/mattermost-server/mlog"
+)
+
+func (s *Server) AutoMergePR() {
+	var prs []*model.PullRequest
+	result := <-s.Store.PullRequest().ListOpen()
+	if result.Err != nil {
+		mlog.Error(result.Err.Error())
+		return
+	}
+	prs = result.Data.([]*model.PullRequest)
+
+	client := NewGithubClient(s.Config.GithubAccessToken)
+	for _, pr := range prs {
+		if !s.isAutoMergeLabelInLabels(pr.Labels) {
+			mlog.Info("No auto merge label for this PR", mlog.Int("pr", pr.Number), mlog.String("repo", pr.RepoName))
+			continue
+		}
+
+		prToMerge, _, err := client.PullRequests.Get(context.Background(), pr.RepoOwner, pr.RepoName, pr.Number)
+		if err != nil {
+			mlog.Error("Error to gettinh the PR info", mlog.Int("pr", pr.Number), mlog.String("repo", pr.RepoName), mlog.Err(err))
+			continue
+		}
+
+		if prToMerge.GetState() == "closed" {
+			continue
+		}
+
+		if prToMerge.GetMergeableState() != "clean" {
+			mlog.Info("PR is not ready to merge", mlog.Int("pr", pr.Number), mlog.String("repo", pr.RepoName))
+			continue
+		}
+
+		// Get the Statuses
+		PRStatus, _, err := client.Repositories.GetCombinedStatus(context.Background(), pr.RepoOwner, pr.RepoName, prToMerge.Head.GetSHA(), nil)
+		if err != nil {
+			mlog.Error("Error to gettinh the PR Status", mlog.Int("pr", pr.Number), mlog.String("repo", pr.RepoName), mlog.Err(err))
+			continue
+		}
+		if prToMerge.Head.GetSHA() != PRStatus.GetSHA() && PRStatus.GetState() != "success" {
+			mlog.Info("PR is not ready to merge", mlog.Int("pr", pr.Number), mlog.String("repo", pr.RepoName))
```

Comment:
This log message is indistinguishable from the one on line 42 -- can we annotate both to call out the specific case that failed, in case any debugging is necessary?

Reply:
yep, will update